src/main/java/org/folio/inventory/storage/external/CollectionResourceRepository.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.inventory.storage.external;
+
+import org.folio.inventory.storage.external.exceptions.ExternalResourceFetchException;
+import org.folio.inventory.support.http.client.OkapiHttpClient;
+import org.folio.inventory.support.http.client.Response;
+
+import java.net.URL;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+public class CollectionResourceRepository extends CollectionResourceClient{
```

Comment:
I'm not sure if we should use extension here.  As I understand the `CollectionResourceRepository ` provides an alternative way of interaction with endpoints, so I'd use delegation here.

Reply:
Agreed, my understanding of the intention of a repository is that it provides a higher level abstraction over the client.

What benefits does inheriting from the client instead of using the client have?